tests/js/server/recovery/collection-count.js

```diff
@@ -0,0 +1,101 @@
+/* jshint globalstrict:false, strict:false, unused: false */
+/* global assertTrue, assertFalse, assertEqual, assertMatch */
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief tests for count recovery
+// /
+// / @file
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2010-2012 triagens GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is triAGENS GmbH, Cologne, Germany
+// /
+// / @author Jan Steemann
+// / @author Copyright 2013, triAGENS GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+const db = require('@arangodb').db;
+const internal = require('internal');
+const jsunity = require('jsunity');
+const fs = require("fs");
+const continueFile = fs.join(fs.getTempPath(), "CONTINUE");
+
+function runSetup () {
+  'use strict';
+  internal.debugClearFailAt();
+
+  db._drop('UnitTestsRecovery');
+  let c = db._create('UnitTestsRecovery');
+
+  internal.debugSetFailAt('RocksDBSettingsManagerSync');
+
+  c.insert({ _key: "test" }, { waitForSync: true });
+  internal.debugTerminate('crashing server');
+}
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief test suite
+// //////////////////////////////////////////////////////////////////////////////
+
+function recoverySuite () {
+  'use strict';
+  jsunity.jsUnity.attachAssertions();
+      
+  return {
+    testRestart1: function () {
+      if (fs.exists(continueFile)) {
+        // skip this phase if we have already been here
+        return;
+      }
+      require("console").info("counts test, phase 1");
+      assertFalse(fs.exists(continueFile));
+
+      let c = db._collection('UnitTestsRecovery');
+      assertEqual(1, c.count());
+      assertEqual(1, c.toArray().length);
+      
+      internal.waitForEstimatorSync();
+
+      fs.write(continueFile, "1");
+      internal.debugTerminate('crashing server');
+    },
```

Comment:
This must be either executed within `runSetup` context or propagate `params.options.disableMonitor = true;` to `js/client/modules/@arangodb/testsuites/recovery.js`. Otherwise WER will catch it as unexpected crash:
```
 [recovery] :     
    during: execution of build\bin\RelWithDebInfo\arangod.exe - -1254: Core dump written; Process facts :
    rootDir: 'c:\vm01-windows\oskar\work\tmp\arangosh_a04560\recovery\2'
    pid: 6920
    exitStatus:
      pid: 6920
      status: ABORTED
      signal: -1254
    coreFilePattern: >-
      c:\jenkins\workspace\arangodb-matrix-pr-windows\EDITION\enterprise\STORAGE_ENGINE\rocksdb\TEST_SUITE\single\limit\windows&&test\core\arangod.exe.6920.dmp
    marking build as crashy.
Marking crashy!
```